src/main/java/com/vinyl/service/UserService.java

```diff
@@ -114,4 +119,25 @@ public CartDetailsDTO getCartDetails(String tokenHash) {
 
         return cartDetails;
     }
+
+    public void deleteProductFromCart(String tokenHash, Integer productId, Integer userId) {
+
+        validatorFactory.getTokenValidator().validate(tokenHash);
+        Token token = tokenRepository.findByHash(tokenHash);
+        User user = token.getUser();
+        if (userId != tokenRepository.findByHash(tokenHash).getUser().getId()) {
```

Comment:
you find the user above, no need for another db call

Reply:
done